ntoskrnl/ps/query.c

```diff
@@ -175,36 +179,50 @@ NtQueryInformationProcess(
             /* Indicate success */
             Status = STATUS_SUCCESS;
 
-            _SEH2_TRY
+            RtlZeroMemory(&QuotaLimits, sizeof(QuotaLimits));
+
+            /* Get max/min working set sizes */
+            QuotaLimits.MaximumWorkingSetSize =
+                Process->Vm.MaximumWorkingSetSize << PAGE_SHIFT;
+            QuotaLimits.MinimumWorkingSetSize =
+                Process->Vm.MinimumWorkingSetSize << PAGE_SHIFT;
+
+            /* Get default time limits */
+            QuotaLimits.TimeLimit.LowPart = MAXULONG;
+            QuotaLimits.TimeLimit.HighPart = MAXULONG;
```

Comment:
I know it's from the old code, but it should be possible to do:
```suggestion
            QuotaLimits.TimeLimit.QuadPart = -1LL;
```

Reply:
Would `MAXLONGLONG` be better than `-1LL` ?

Reply:
This should work, of course; I just suggested the -1 because this is what's being used for the other fields.

Reply:
`-1LL` is actually correct. Checked on Win2k3.
![NTQIP_Win2k3_2](https://github.com/user-attachments/assets/72037fb7-c2ac-48f2-93eb-0153c077dc77)